### ISSUE TITLE
HTTP error. - Uploading Media Files Windows

### ISSUE THREAD

**Issue by DamirPecnik** [timber/timber#1346]
Trying to upload image to media library but getting HTTP error.

Whoops \ Exception \ ErrorException (E_WARNING)
preg_match(): Compilation failed: unknown property name after \P or \p at offset 4

See screenshot of the error.
![2017-03-08 09 55 10](https://cloud.githubusercontent.com/assets/10489911/23698132/ab24f888-03ea-11e7-9f5d-3b2fc42a9ac2.jpg)

### What version of WordPress, PHP and Timber are you using?
WordPress 4.7.3, PHP 7.0.13, Timber 1.2.4 - Windows 10 - Mamp Pro

### How did you install Timber? (for example, from GitHub, Composer/Packagist, WP.org?)
Part of the Gantry 5 Plugin

@mahagr will have some additional info

**Comment by mahagr** [comment]
Looks like Timber is not properly escaping `$filename`, `$regexdir` and `$ext`, causing regexp to have `F:\P...` in it.

Please use `preg_quote($var, '/')` for all untrusted variables, which are not meant to be regular expressions.

**Comment by mahagr** [comment]
Here is the fix:

```php
	protected static function process_delete_generated_files( $filename, $ext, $dir, $search_pattern, $match_pattern = null ) {
		$searcher = '/'.$filename.$search_pattern;
		foreach ( glob($dir.$searcher) as $found_file ) {
			$pattern = '/'.preg_quote($dir, '/').'\/'.preg_quote($filename, '/').$match_pattern.preg_quote($ext, '/').'/';
			$match = preg_match($pattern, $found_file);
			if ( !$match_pattern || $match ) {
				unlink($found_file);
			}
		}
	}
```

Note that I didn't test it. :)

**Comment by jarednova** [comment]
@mahagr could you submit this as a PR?

**Comment by mahagr** [comment]
There you are. Looks like I did the change against my master, but yeah, I can re-branch that later. :)

**Comment by jarednova** [comment]
Closed by #1348
